src/datachain/lib/convert/values_to_tuples.py

```diff
@@ -71,7 +71,10 @@ def values_to_tuples(  # noqa: C901, PLR0912
                     f"signal '{k}' has unsupported type '{typ.__name__}'."
                     f" Please use DataModel types: {DataTypeNames}",
                 )
-            types_map[k] = typ
+            if typ is list:
+                types_map[k] = list[type(v[0][0])]  # type: ignore[misc]
+            else:
+                types_map[k] = typ
```

Comment:
WDYT about moving this whole type mapping logic into a small helper function? `values_to_tuples()` is getting long, and it would allow for doing the list check recursively so this doesn't break for deeper nested lists.

Reply:
yeah, this function needs redesign. I'd do that as a separate PR.